File: apps/mobile/modules/quickpose-camera/android/src/main/java/expo/modules/quickposecamera/QuickPoseCameraModule.kt
package expo.modules.quickposecamera

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class QuickPoseCameraModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("QuickPoseCamera")

    View(QuickPoseCameraView::class) {
      Events("onUpdate")
      Prop("sdkKey") { _: QuickPoseCameraView, _: String -> }
      Prop("features") { _: QuickPoseCameraView, _: List<String> -> }
      Prop("featureStyles") { _: QuickPoseCameraView, _: Map<String, Any>? -> }
      Prop("useFrontCamera") { _: QuickPoseCameraView, _: Boolean -> }
    }
  }
}

File: apps/mobile/modules/quickpose-camera/ios/QuickPoseCameraModule.swift
import ExpoModulesCore

public class QuickPoseCameraModule: Module {
  public func definition() -> ModuleDefinition {
    Name("QuickPoseCamera")

    View(HealthyQuickPoseView.self) {
      Events("onUpdate")

      Prop("sdkKey") { (view: HealthyQuickPoseView, sdkKey: String) in
        view.sdkKey = sdkKey
      }

      Prop("features") { (view: HealthyQuickPoseView, features: [String]) in
        view.featureKeys = features
      }

      Prop("featureStyles") { (view: HealthyQuickPoseView, styles: [String: Any]?) in
        view.featureStyles = styles ?? [:]
      }

      Prop("useFrontCamera") { (view: HealthyQuickPoseView, useFrontCamera: Bool) in
        view.useFrontCamera = useFrontCamera
      }
    }
  }
}
